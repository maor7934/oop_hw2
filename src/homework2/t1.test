# Very long test

CreateNode n1 10
CreateNode n2 10
CreateNode n3 10
CreateNode n4 10
CreateNode n5 10
CreateNode n6 10
CreateNode n7 10
CreateNode n8 10
CreateNode n9 10

CreateNode n10 1
CreateNode n11 1
CreateNode n12 1
CreateNode n13 1
CreateNode n14 1
CreateNode n15 1
CreateNode n16 1
CreateNode n17 1
CreateNode n18 1
CreateNode n19 1

CreateNode n21 100
CreateNode n22 100
CreateNode n23 100
CreateNode n24 100
CreateNode n25 100
CreateNode n26 100
CreateNode n27 100
CreateNode n28 100
CreateNode n29 100

CreateNode n30 500

CreateGraph A
AddNode A n1
AddNode A n10
AddNode A n20
AddNode A n30
AddNode A n2
AddNode A n12
AddNode A n13
AddNode A n14
AddNode A n25
AddNode A n26
AddNode A n19

AddEdge A n1 n30
AddEdge A n10 n30
AddEdge A n20 n30
AddEdge A n2 n30
AddEdge A n12 n30
AddEdge A n13 n30
AddEdge A n14 n30
AddEdge A n25 n30
AddEdge A n25 n30
AddEdge A n26 n30
AddEdge A n19 n30

AddEdge A n30 n10
AddEdge A n30 n20
AddEdge A n30 n2




AddEdge A n1 n2
AddEdge A n1 n14
AddEdge A n25 n13
AddEdge A n24 n25
AddEdge A n30 n19
AddEdge A n19 n26
AddEdge A n26 n25
AddEdge A n25 n24
AddEdge A n13 n26
AddEdge A n24 n10
AddEdge A n19 n1
AddEdge A n1 n19
AddEdge A n24 n20
AddEdge A n20 n10
AddEdge A n10 n14

ListNodes A
ListChildren A n1
ListChildren A n30
ListChildren A n25

# Find a path
FindPath graph1 n1 -> n30
FindPath graph1 n30 -> n1

FindPath graph1 n1 -> n24
FindPath graph1 n19 -> n1

FindPath graph1 n10 -> n26




