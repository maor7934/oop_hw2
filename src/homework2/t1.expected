# Very long test

created node n1 with cost 10
created node n2 with cost 10
created node n3 with cost 10
created node n4 with cost 10
created node n5 with cost 10
created node n6 with cost 10
created node n7 with cost 10
created node n8 with cost 10
created node n9 with cost 10

created node n10 with cost 1
created node n11 with cost 1
created node n12 with cost 1
created node n13 with cost 1
created node n14 with cost 1
created node n15 with cost 1
created node n16 with cost 1
created node n17 with cost 1
created node n18 with cost 1
created node n19 with cost 1
created node n20 with cost 27

created node n21 with cost 100
created node n22 with cost 100
created node n23 with cost 100
created node n24 with cost 100
created node n25 with cost 100
created node n26 with cost 100
created node n27 with cost 100
created node n28 with cost 100
created node n29 with cost 100

created node n30 with cost 500

created graph A
added node n1 to A
added node n10 to A
added node n20 to A
added node n30 to A
added node n2 to A
added node n12 to A
added node n13 to A
added node n14 to A
added node n25 to A
added node n26 to A
added node n19 to A

added edge from n1 to n30 in A
added edge from n10 to n30 in A
added edge from n20 to n30 in A
added edge from n2 to n30 in A
added edge from n12 to n30 in A
added edge from n13 to n30 in A
added edge from n14 to n30 in A
added edge from n25 to n30 in A
Exception: homework2.CommandException: Graph A already have edge between nodes n25 -> n30
added edge from n26 to n30 in A
added edge from n19 to n30 in A

added edge from n30 to n10 in A
added edge from n30 to n20 in A
added edge from n30 to n2 in A




added edge from n1 to n2 in A
added edge from n1 to n14 in A
added edge from n25 to n13 in A
Exception: homework2.CommandException: Graph A doesn't have node n24 \ n25
added edge from n30 to n19 in A
added edge from n19 to n26 in A
added edge from n26 to n25 in A
Exception: homework2.CommandException: Graph A doesn't have node n25 \ n24
added edge from n13 to n26 in A
Exception: homework2.CommandException: Graph A doesn't have node n24 \ n10
added edge from n19 to n1 in A
added edge from n1 to n19 in A
Exception: homework2.CommandException: Graph A doesn't have node n24 \ n20
added edge from n20 to n10 in A
added edge from n10 to n14 in A

A contains: n1 n10 n12 n13 n14 n19 n2 n20 n25 n26 n30
the children of n1 in A are: n14 n19 n2 n30
the children of n30 in A are: n10 n19 n2 n20
the children of n25 in A are: n13 n30

# Find a path
shortest path in A: n1 n30
shortest path in A: n30 n19 n1

no path found in A
shortest path in A: n19 n1

shortest path in A: n10 n30 n19 n26




